### ISSUE TITLE
Driver Drowsiness Detection using CNN and OpenCV

### ISSUE THREAD

**Issue by keshripritesh** [PRIYANSHU2026/DataSentience-AIML#97]
---

### 📍 Project Title:

**Driver Drowsiness Detection using CNN and OpenCV**

---

### 📖 Description:

Drowsiness while driving is one of the leading causes of road accidents. This project aims to develop a **real-time driver drowsiness detection system** using **OpenCV and a Convolutional Neural Network (CNN)**. The system monitors the driver’s eyes using a webcam and triggers an alert if eyes remain closed for more than 15 continuous frames.

By combining traditional **Haar Cascade eye detection** with a trained CNN model for eye state classification (open/closed), the system provides accurate and real-time feedback to help reduce drowsy driving risks.

This project is beginner-friendly and ideal for contributors interested in **deep learning, computer vision, and real-time alert systems**.

---

### 🔧 Tech Stack:

* **Languages**: Python

* **Libraries**:

  * TensorFlow
  * Keras
  * OpenCV
  * NumPy

* **Tools**:

  * Jupyter / Python3
  * Haarcascade XML classifiers
  * Webcam or external camera

---

### 📂 Folder Structure:

```
Driver-Drowsiness-Detection/
├── model/
│   └── cnn_eye_state_model.h5
├── haarcascade/
│   ├── haarcascade_frontalface_alt.xml
│   └── haarcascade_eye_tree_eyeglasses.xml
├── detection.py
├── train_model.py
├── dataset/
├── README.md
└── requirements.txt
```

---

### 🚀 How to Run

1. Clone the repo:

```bash
git clone https://github.com/<your-username>/Driver-Drowsiness-Detection.git
cd Driver-Drowsiness-Detection
```

2. Install dependencies:

```bash
pip install -r requirements.txt
```

3. Run the detection:

```bash
python detection.py
```

Make sure your webcam is connected and the model is trained and saved as `cnn_eye_state_model.h5`.

---

### ✅ Features

* 🔍 Detects real-time face and eye landmarks
* 🧠 Classifies eye state (open/closed) using CNN
* ⏰ Triggers an alarm if eyes remain closed for 15+ frames
* 📸 Uses Haarcascade for fast face and eye detection
* 🛡️ Potential use in **driver monitoring systems**

---

### 📷 Demo

![Demo](https://user-images.githubusercontent.com/73430464/160766013-f7214357-ecf7-4463-b43f-077c45936da2.png)

---

### 🧩 Future Scope / First Issues

* [ ] Add GUI to toggle alarm sensitivity
* [ ] Integrate sound alert with volume control
* [ ] Train with a larger dataset for improved accuracy
* [ ] Add eye tracking support with dlib or Mediapipe
* [ ] Extend to detect yawning or head-nod patterns

---

### 🙋 Who Can Contribute?

* ML/DL enthusiasts interested in model tuning
* OpenCV beginners for real-time image processing
* Web/Python developers to improve usability
* UI/UX contributors for dashboard integration

---

### 📬 Final Note:

This project is open-source and ready for collaborative contributions. I'm excited to submit it for Social Summer of Code and look forward to mentoring fellow contributors who want to explore applied deep learning and safety-tech.

---
